### ISSUE TITLE
Conversion from logits to probabilities happens on a batch by batch basis

### ISSUE THREAD

**Issue by GabrielBianconi** [Lightning-AI/torchmetrics#2195]
## 🐛 Bug

I'm using `BinaryAUROC` and calling `.update(logits, y)` after each batch (`validation_step`), expecting the following behavior:

"If preds has values outside [0,1] range we consider the input to be logits and will auto apply sigmoid per element."

I noticed some discrepancies in the results, and after investigating realized that the rule above happens on a batch by batch basis rather than across the entire dataset (it happens in `_binary_precision_recall_curve_format`, which is called by `update`). 

Therefore, if some batch has all logits in [0, 1], then the conversion doesn't happen for that batch.

My use case has batch size = 1, so this happens often.

### Expected behavior

I'd expect that the conversion would happen consistently across the entire dataset (i.e., when calling `.compute()`). If not that, this behavior should be documented more prominently.

### Environment

```
torchmetrics==1.0.3
```

**Comment by ocharles** [comment]
I came to report just this. As a "workaround", I always call `sigmoid` first, as at least that path is sound.

**Comment by ocharles** [comment]
Dupe of https://github.com/Lightning-AI/torchmetrics/issues/1604, probably

**Comment by kushagragpt99** [comment]
Hey! Wanted to see if this issue is still open. I saw that #1676 was meant to solve this but I’m not sure if that was merged. Happy to contribute :)

**Comment by Borda** [comment]
> Hey! Wanted to see if this issue is still open. I saw that [#1676](https://github.com/Lightning-AI/torchmetrics/pull/1676) was meant to solve this but I’m not sure if that was merged. Happy to contribute :)

not yet...
cc: @SkafteNicki

**Comment by mohansree14** [comment]
Hi! I'd like to work on this if it's still open. To keep the change scoped, I'm planning to start with clearly documenting the per-batch conversion behavior in the relevant docstring(s) (e.g. `BinaryAUROC`/`update()`) rather than changing the update-time logic itself, since that could affect other metrics relying on the current batching behavior. Happy to go the behavior-change route instead if you'd prefer — let me know. This would be my first contribution here, thanks!
